### PR TITLE
Leverage libcudacxx limits and cstdint defines 

### PR DESCRIPTION
**Pull request by robertmaynard**
Instead of using hardcoded defines in jitify generated content, leverage the defines that exist in libcudacxx

This removes compilation errors when the defines between jitify and libcudacxx differ ( e.g. `LLONG_MAX  0x7fffffffffffffff` vs `LLONG_MAX  0x7fffffffffffffffLL` ).

Plus it removes the upkeep burden of jitify of having to stay in sync with header changes

### CODE REVIEW

File: jitify2.hpp [hunk 1]
```diff
@@ -3673,11 +3616,7 @@ typedef unsigned short uint_least16_t;
 typedef unsigned int uint_least32_t;
 typedef unsigned long long uint_least64_t;
 typedef unsigned long long uintmax_t;
-#if defined _WIN32 || defined _WIN64
-typedef unsigned long long uintptr_t;  // optional
-#else  // not Windows
-typedef unsigned long uintptr_t;  // optional
-#endif
+
```
Comment (bdice) [inline jitify2.hpp]: I'd remove the blank lines for consistency with the previous densely-packed code.
```suggestion
```

File: jitify2.hpp [hunk 2]
```diff
@@ -3659,7 +3602,7 @@ typedef signed short int_least16_t;
 typedef signed int int_least32_t;
 typedef signed long long int_least64_t;
 typedef signed long long intmax_t;
-typedef signed long intptr_t;  // optional
+
```
Comment (bdice) [inline jitify2.hpp]:
```suggestion
```

**Review by bdice — APPROVED** [review]
This looks good to me with one minor comment about removing blank lines. I found the places in libcudacxx where these macros and typedefs are defined, and it seems like this should be valid.

**Comment by bdice** [comment]
@robertmaynard Are we planning to upstream these changes, too?

**Comment by robertmaynard** [comment]
> @robertmaynard Are we planning to upstream these changes, too?

Yes there are plans to upstream this. Currently working on if these changes need to be conditional based on the CUDA version ( support < cuda 11 )

**Review by shwina — APPROVED** [review]
This LGTM from a cuDF standpoint!

**Pull request merged**